### DISCUSSION TITLE
Locating on-chain transaction history

### DISCUSSION THREAD

**Discussion by gus4rs** [lightningnetwork/lnd#8760] [General]
I have restored the on-chain ```lnd``` wallet from the seed and passphrase, then I generated the vpub (I am on testnet)

```lncli wallet accounts list  --name default```

and took ```vpub``` and imported into Sparrow wallet, and I can see all transactions I used to fund and open channels

But I cannot see those transactions when doing:

```lncli listchaintxns```

What am I missing?

**Comment by guggero** [comment]
Is the `lnd` wallet fully synced to chain? After restoring from seed it will re-scan the whole chain which can take a long time (and with the current block storms on testnet even more). Also, if you interrupt the re-scan with a restart of the node it will not continue automatically, so you should let it finish when you do the recovery (checking the log for the status). If you did interrupt the re-scan then you need to start the node with the `--reset-wallet-transactions` flag (or `reset-wallet-transactions=true` in the config file) to trigger a new re-scan.

**Reply by gus4rs** [reply]
I recovered with ```lncli create``` and have a script to wait for both ```.synced_to_chain``` and ```.synced_to_graph``` to true

**Reply by guggero** [reply]
Did you specify a non-zero recovery window?

**Reply by gus4rs** [reply]
I (the recover script) just accepted default value. I tried to do a ```reset-wallet-transactions=true``` but it didn't go back enough and took 2min

```
May 15 12:08:35 localhost.localdomain lnd[2197020]: 2024-05-15 12:08:35.378 [INF] LNWL: Started rescan from block 00000000049d86890e6e272112e74263aeef60e60146055f0736d75768934462 (height 2813042) for 2 addresses
[INF] LNWL: Catching up block hashes to height 2815472, this might take a while
[INF] LNWL: Done catching up block hashes
[INF] LNWL: Finished rescan for 2 addresses (synced to block 0000000000000011d4f13a8ff4999f6088e19ce095b856753025c5b1ed4209b3, height 2815472)

```

**Reply by guggero** [reply]
How/where did you create the original seed? Sounds like it doesn't properly encode the wallet's birthday then.

**Reply by gus4rs** [reply]
I didn't create it. `lnd` generated with `lncli create`

**Reply by guggero** [reply]
Can you do a fresh restore and post the full log please?

**Comment by gus4rs** [comment]
I found the issue. The script was stopping the process right after the call to create, and the recover was interrupted.
